src/main/java/org/jitsi/jigasi/transcription/WhisperingTranscriptionService.java

```diff
@@ -0,0 +1,376 @@
+
+        /* Collect results*/
+        private StringBuilder result;
+
+        WhisperingWebsocketSession(TranscriptionRequest request) {
+            this.closeLatch = new CountDownLatch(1);
+            this.request = request;
+            this.result = new StringBuilder();
+        }
+
+        @OnWebSocketClose
+        public void onClose(int statusCode, String reason) {
+            this.closeLatch.countDown(); // trigger latch
+        }
+
+        @OnWebSocketConnect
+        public void onConnect(Session session) {
+            try
+            {
+                AudioFormat format = request.getFormat();
+                WhisperingContext ctx = new WhisperingContext(0.0);
+                session.getRemote().sendString(ctx.toJSON().toString());
+                ByteBuffer audioBuffer = ByteBuffer.wrap(request.getAudio());
+                session.getRemote().sendBytes(audioBuffer);
+                session.getRemote().sendString(EOF_MESSAGE);
+            }
+            catch (IOException e)
+            {
+                logger.error("Error to transcribe audio", e);
+            }
+        }
+
+        @OnWebSocketMessage
+        public void onMessage(String msg) {
+            result.append(msg);
+            result.append('\n');
+        }
+
+        @OnWebSocketError
+        public void onError(Throwable cause) {
+            logger.error("Websocket connection error", cause);
+        }
+
+        public String getResult() {
+            return result.toString();
+        }
+
+        void awaitClose()
+                throws InterruptedException {
+            closeLatch.await();
+        }
+    }
+
+    /**
+     * Represent the Whispering Context to instantiate the transcription
+     * service.
+     */
+    public class WhisperingContext {
+        /* Starting timestamp of the transcription service. */
+        double timestamp;
+
+        WhisperingContext(double timestamp) {
+            this.timestamp = timestamp;
+        }
+
+        public JSONObject toJSON() {
+            JSONObject ctx = new JSONObject();
+            ctx.put("timestamp", this.timestamp);
+            ctx.put("buffer_tokens", new JSONArray());
+            ctx.put("buffer_mel", JSONObject.NULL);
+            ctx.put("vad", true);
+            JSONArray temperatures = new JSONArray();
+            for (int i = 0; i <= 10; i += 2) {
+                temperatures.add(i / 10);
+            }
+            ctx.put("temperatures", temperatures);
+            ctx.put("allow_padding", true);
+            ctx.put("patience", JSONObject.NULL);
+            ctx.put("compression_ratio_threshold", 2.4);
+            ctx.put("logprob_threshold", -1.0);
+            ctx.put("no_captions_threshold", 0.6);
+            ctx.put("best_of", 5);
+            ctx.put("beam_size", 5);
+            ctx.put("no_speech_threshold", 0.6);
+            ctx.put("buffer_threshold", 0.5);
+            ctx.put("vad_threshold", 0.5);
+            ctx.put("data_type", "float32");
```

Comment:
if using GPU, I expect it should be faster with float16

Reply:
My bad this was supposed to be int64, as, from my understanding, this is the audio format Jigasi sends.

I have implemented a convertion to float32 in https://github.com/shirayu/whispering/pull/36 but I will suggest float16 for better performances.